### ISSUE TITLE
Breaking(?) change from Behat/Gherkin should be outlined in changelog?

### ISSUE THREAD

**Issue by nickvergessen** [Behat/Behat#1514]
Behat/Behat 3.15.0 bumped Behat/Gherkin
The "simple" change of https://github.com/Behat/Gherkin/commit/5a0836db6fd4f0d6fe1db6dfff5118ab615681b4 however breaks all TableNodes that use JSON data with escaped backslashes.

In our case we had:
```gherkin
    And user "admin" creates global flow with 200
      | name      | Admin flow                       |
      | class     | OCA\FilesAccessControl\Operation |
      | entity    | OCA\WorkflowEngine\Entity\File   |
      | events    | []                               |
      | operation | deny                             |
      | checks-0  | {"class":"OCA\\WorkflowEngine\\Check\\FileName","operator":"is","value":"foobar.txt"} |
```

working for years, and the above change makes touch all tables and escape the slashes another time
```gherkin
    And user "admin" creates global flow with 200
      | name      | Admin flow                       |
      | class     | OCA\FilesAccessControl\Operation |
      | entity    | OCA\WorkflowEngine\Entity\File   |
      | events    | []                               |
      | operation | deny                             |
      | checks-0  | {"class":"OCA\\\\WorkflowEngine\\\\Check\\\\FileName","operator":"is","value":"foobar.txt"} |
```

- We can of course do that, but I think it should at least be added to the changelog for clarity?
There already was another question about it in https://github.com/Behat/Gherkin/issues/268
But the library issues/changelog might not be what most users check.

**Comment by carlos-granados** [comment]
Thanks @nickvergessen your suggestion to mention this in the changelog seems reasonable. @acoulton how should we handle this?

**Comment by nickvergessen** [comment]
*My suggestion would be to add a "Breaking" section to https://github.com/Behat/Behat/blob/master/CHANGELOG.md#3150---2024-10-29 and manually edit it also into the existing tag/release description (without a further release for now)*

**Comment by acoulton** [comment]
Hmm - I'm not sure. 

We could flag it in the behat changelog (I hadn't actually noticed that the minimum requirement was bumped, as it was done direct on master before we took over).

However, ultimately the break is in behat/gherkin@4.10.0 and will affect people even if they stick with behat/behat@3.14.0.

So the main / correct place to highlight it would be the gherkin CHANGELOG (where it is also not clear). It's only coincidence that both packages are from the same org, it could just as easily be an external dependency causing the problem and I don't really want a convention that the behat changelog details the changes in other packages as that will be impossible to maintain and will give people a false sense of security if we miss something that affects them.

tl;dr - if people are upgrading behat/gherkin then they *should* be checking the gherkin changes, even if they're upgrading behat at the same time. But we definitely need to fix the gherkin changelog asap.

**Comment by stof** [comment]
The Behat changelog should highlight the fact that it migrates to a new major version of behat/gherkin though.

**Comment by acoulton** [comment]
@stof it doesn't, it bumps from `^4.9.0` to `^4.10.0` https://github.com/Behat/Behat/commit/382826ef35ee94c03a968d370f8878ea63335767

**Comment by nickvergessen** [comment]
So it doesn't bump a major, but a minor. Might still be worth to outline, to give users a hint for related changelogs :)

**Comment by acoulton** [comment]
Added a hint in #1517

**Comment by nickvergessen** [comment]
Thanks a lot everyone :+1:
